application_sandbox/conditional_rendering/main.cpp

```diff
@@ -0,0 +1,528 @@
+        app()->swapchain().height(),                // height
+        1                                           // layers
+    };
+
+    ::VkFramebuffer raw_framebuffer;
+    app()->device()->vkCreateFramebuffer(
+        app()->device(), &framebuffer_create_info, nullptr, &raw_framebuffer);
+    frame_data->framebuffer_ = containers::make_unique<vulkan::VkFramebuffer>(
+        data_->allocator(),
+        vulkan::VkFramebuffer(raw_framebuffer, nullptr, &app()->device()));
+
+    (*frame_data->command_buffer_)
+        ->vkBeginCommandBuffer((*frame_data->command_buffer_),
+                               &sample_application::kBeginCommandBuffer);
+    vulkan::VkCommandBuffer& cmdBuffer = (*frame_data->command_buffer_);
+
+    VkRenderPassBeginInfo pass_begin = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO,  // sType
+        nullptr,                                   // pNext
+        *render_pass_,                             // renderPass
+        *frame_data->framebuffer_,                 // framebuffer
+        {{0, 0},
+         {app()->swapchain().width(),
+          app()->swapchain().height()}},  // renderArea
+        0,                                // clearValueCount
+        nullptr                           // clears
+    };
+
+    // Two colorful cubes with a blue background
+    VkConditionalRenderingBeginInfoEXT conditional_begin1 = {
+        VK_STRUCTURE_TYPE_CONDITIONAL_RENDERING_BEGIN_INFO_EXT,
+        nullptr,
+        conditional_data_->get_buffer(),
+        conditional_data_->get_offset_for_frame(frame_index),
+        0,
+    };
+
+    // Single black cube with a pink bacground
+    VkConditionalRenderingBeginInfoEXT conditional_begin2 = conditional_begin1;
+    conditional_begin2.flags = VK_CONDITIONAL_RENDERING_INVERTED_BIT_EXT;
+
+    VkBufferMemoryBarrier to_use_in_comp{
+        VK_STRUCTURE_TYPE_BUFFER_MEMORY_BARRIER,
+        nullptr,
+        VK_ACCESS_HOST_WRITE_BIT,
+        VK_ACCESS_SHADER_READ_BIT | VK_ACCESS_SHADER_WRITE_BIT,
+        VK_QUEUE_FAMILY_IGNORED,
+        VK_QUEUE_FAMILY_IGNORED,
+        dispatch_data_->get_buffer(),
+        dispatch_data_->get_offset_for_frame(frame_index),
+        dispatch_data_->aligned_data_size()};
+    VkBufferMemoryBarrier to_use_in_frag{
+        VK_STRUCTURE_TYPE_BUFFER_MEMORY_BARRIER,
+        nullptr,
+        VK_ACCESS_SHADER_READ_BIT | VK_ACCESS_SHADER_WRITE_BIT,
+        VK_ACCESS_SHADER_READ_BIT,
+        VK_QUEUE_FAMILY_IGNORED,
+        VK_QUEUE_FAMILY_IGNORED,
+        dispatch_data_->get_buffer(),
+        dispatch_data_->get_offset_for_frame(frame_index),
+        dispatch_data_->aligned_data_size()};
+
+    cmdBuffer->vkCmdPipelineBarrier(cmdBuffer, VK_PIPELINE_STAGE_HOST_BIT,
+                                    VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT, 0, 0,
+                                    nullptr, 1, &to_use_in_comp, 0, nullptr);
+
+    cmdBuffer->vkCmdBeginConditionalRenderingEXT(cmdBuffer,
```

Comment:
Can we describe in the sample description what each conditional block is doing?
It  seemed to me (from the readme) that we would conditionally render one of two scenes, but this dispatch is also in a conditional block. Would be good to understand how that is used.

Maybe just (in the readme for the sample, or in a comment block above the command recording) a simple high-level description of what/how things are being set up.

```
CommandBuffer
   - Pipeline barriers
   - ConditionalBegin  // What is the condition we care about
       - Binding
       - Dispatch // What compute are we using
   - Barriers
   - RenderPass
        - Binding
        - 
....
```

Reply:
Added description in README